src/handle-initial-jsmd.js

```diff
@@ -27,4 +27,27 @@ export default async function handleInitialJsmd(store) {
       store.dispatch(evaluateAllCells())
     }
   }
+
+  state = store.getState();
+  if (state.viewMode === 'EXPLORE_VIEW' && state.userData.name !== state.ownerData.username) {
+    if (state.userData.name === undefined) {
+      store.dispatch(addHeaderMessage(
+        'To save to the server, you need to ',
+        'login.',
+        () => {
+          store.dispatch(login())
+          return true
+        },
```

Comment:
This doesn't seem right to me -- it should be up to the react component to display these messages. In this case I think we can just detect that the user is not logged in and display this kind of message unconditionally. Same idea for the one below -- display the banner if the currently loaded notebook is not owned by the current user.

Reply:
+1